lib/Doctrine/ORM/Proxy/ProxyFactory.php

```diff
@@ -71,10 +76,11 @@ class ProxyFactory extends AbstractProxyFactory
      * @param string                 $proxyNs      The namespace to use for the proxy classes.
      * @param boolean|int            $autoGenerate The strategy for automatically generating proxy classes. Possible
      *                                             values are constants of Doctrine\Common\Proxy\AbstractProxyFactory.
+     * @param int                    $umask        The umask that controls file permissions.
      */
-    public function __construct(EntityManagerInterface $em, $proxyDir, $proxyNs, $autoGenerate = AbstractProxyFactory::AUTOGENERATE_NEVER)
+    public function __construct(EntityManagerInterface $em, $proxyDir, $proxyNs, $autoGenerate = AbstractProxyFactory::AUTOGENERATE_NEVER, $umask = 0002)
     {
-        $proxyGenerator = new ProxyGenerator($proxyDir, $proxyNs);
+        $proxyGenerator = new ProxyGenerator($proxyDir, $proxyNs, $umask);
```

Comment:
Any 	recommendations how to get the fix in v2.5.5 without breaking compatibility? I consider this more like a bug fix than a new feature given the nature of https://github.com/doctrine/common/issues/710.